saga/saga_state.go

```diff
@@ -246,24 +246,20 @@ func (state *SagaState) addTaskData(taskId string, msgType SagaMessageType, data
 }
 
 /*
- * Applies the supplied message to the supplied sagaState.  Does not mutate supplied Saga State
- * Instead returns a new SagaState which has the update applied to it
+ * Applies the supplied message to the supplied sagaState.
+ * Mutates state only if msg is a valid transition.
```

Comment:
Is this true?  Couldn't it mutate and return an error?  That was the reason for the copy in the first place